cli: TestZipIncludeAndExcludeFilesDataDriven failed

cli.TestZipIncludeAndExcludeFilesDataDriven [failed](https://tanzanite.cluster.engflow.com/invocations/default/9a7fb641-25c9-4d78-88b1-c3e31766e292?testReportRun=3&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on release-25.2.0-rc @ [22cec2232cf3ab2a0e5f93cd35aa1217d62a4400](https://github.com/cockroachdb/cockroach/commits/22cec2232cf3ab2a0e5f93cd35aa1217d62a4400):


```
debug/system.migrations.txt
debug/system.namespace.txt
debug/system.privileges.txt
debug/system.protected_ts_meta.txt
debug/system.protected_ts_records.txt
debug/system.rangelog.txt
debug/system.replication_constraint_stats.txt
debug/system.replication_critical_localities.txt
debug/system.replication_stats.txt
debug/system.reports_meta.txt
debug/system.role_id_seq.txt
debug/system.role_members.txt
debug/system.role_options.txt
debug/system.scheduled_jobs.txt
debug/system.settings.txt
debug/system.span_configurations.txt
debug/system.sql_instances.txt
debug/system.sql_stats_cardinality.txt
debug/system.sqlliveness.txt
debug/system.statement_diagnostics.txt
debug/system.statement_diagnostics_requests.txt
debug/system.statement_statistics_limit_5000.txt
debug/system.table_statistics.txt
debug/system.task_payloads.txt
debug/system.tenant_settings.txt
debug/system.tenant_tasks.txt
debug/system.tenant_usage.txt
debug/system.tenants.txt
debug/system.zones.txt
debug/tenant_ranges.err.txt
    zip_test.go:1455: -- test log scope end --
    zip_test.go:1393: 
          testdata/zip/file-filters/testzip_file_filters:2:
          --include-files=*
        output didn't match expected:
          @@ -35,11 +35,12 @@
           debug/events.json
           debug/hot-ranges-tenant.sh
           debug/hot-ranges.sh
           debug/liveness.json
           debug/nodes.json
          -debug/nodes/1/cpu.pprof
          +debug/nodes/1/cpu.pprof.err.txt
          +debug/nodes/1/cpuprof/cpuprof.2025-05-27T08_51_48.726.70.pprof
           debug/nodes/1/crdb_internal.active_range_feeds.txt
           debug/nodes/1/crdb_internal.cluster_replication_node_stream_checkpoints.txt
           debug/nodes/1/crdb_internal.cluster_replication_node_stream_spans.txt
           debug/nodes/1/crdb_internal.cluster_replication_node_streams.txt
           debug/nodes/1/crdb_internal.feature_usage.txt
--- FAIL: TestZipIncludeAndExcludeFilesDataDriven/testzip_file_filters (81.50s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server @cockroachdb/obs-india-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestZipIncludeAndExcludeFilesDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51000

--- a/pkg/cli/testdata/zip/file-filters/testzip_file_filters
+++ b/pkg/cli/testdata/zip/file-filters/testzip_file_filters
@@ -1,5 +1,5 @@
-#include all filters
---include-files=*
+#include all filters. We are excluding cpu.pprof file to avoid flakiness in tests.
+--include-files=* --exclude-files=cpu.pprof
 ----
 debug/crdb_internal.cluster_contention_events.txt
 debug/crdb_internal.cluster_database_privileges.txt
@@ -40,7 +40,6 @@ debug/hot-ranges-tenant.sh
 debug/hot-ranges.sh
 debug/liveness.json
 debug/nodes.json
-debug/nodes/1/cpu.pprof
 debug/nodes/1/crdb_internal.active_range_feeds.txt
 debug/nodes/1/crdb_internal.cluster_replication_node_stream_checkpoints.txt
 debug/nodes/1/crdb_internal.cluster_replication_node_stream_spans.txt
@@ -290,8 +289,8 @@ debug/hot-ranges.sh
 debug/pprof-summary.sh
 debug/tenant_ranges.err.txt
 
-#exclude only log files
---exclude-files=*.log
+#exclude only log files. We are excluding cpu.pprof file to avoid flakiness in tests.
+--exclude-files=*.log,cpu.pprof
 ----
 debug/crdb_internal.cluster_contention_events.txt
 debug/crdb_internal.cluster_database_privileges.txt
@@ -332,7 +331,6 @@ debug/hot-ranges-tenant.sh
 debug/hot-ranges.sh
 debug/liveness.json
 debug/nodes.json
-debug/nodes/1/cpu.pprof
 debug/nodes/1/crdb_internal.active_range_feeds.txt
 debug/nodes/1/crdb_internal.cluster_replication_node_stream_checkpoints.txt
 debug/nodes/1/crdb_internal.cluster_replication_node_stream_spans.txt
@@ -412,8 +410,8 @@ debug/system.zones.txt
 debug/tenant_ranges.err.txt
 
 
-#exclude only txt files
---exclude-files=*.txt
+#exclude only txt files. We are excluding cpu.pprof file to avoid flakiness in tests.
+--exclude-files=*.txt,cpu.pprof
 ----
 debug/debug_zip_command_flags.txt
 debug/events.json
@@ -421,7 +419,6 @@ debug/hot-ranges-tenant.sh
 debug/hot-ranges.sh
 debug/liveness.json
 debug/nodes.json
-debug/nodes/1/cpu.pprof
 debug/nodes/1/details.json
 debug/nodes/1/gossip.json
 debug/nodes/1/heap.pprof
@@ -435,8 +432,8 @@ debug/settings.json
 debug/tenant_ranges.err.txt
 
 
-#exclude only json files
---exclude-files=*.json
+#exclude only json files. We are excluding cpu.pprof file to avoid flakiness in tests.
+--exclude-files=*.json,cpu.pprof
 ----
 debug/crdb_internal.cluster_contention_events.txt
 debug/crdb_internal.cluster_database_privileges.txt
@@ -474,7 +471,6 @@ debug/crdb_internal.zones.txt
 debug/debug_zip_command_flags.txt
 debug/hot-ranges-tenant.sh
 debug/hot-ranges.sh
-debug/nodes/1/cpu.pprof
 debug/nodes/1/crdb_internal.active_range_feeds.txt
 debug/nodes/1/crdb_internal.cluster_replication_node_stream_checkpoints.txt
 debug/nodes/1/crdb_internal.cluster_replication_node_stream_spans.txt

